libraries/support/azure/service.rb

```diff
@@ -42,13 +47,22 @@ def to_struct(data)
       return data unless data.is_a?(Hash)
       return data if data.empty?
 
-      ResponseStruct.create(data.keys.map(&:to_sym), data.values.map { |v| to_struct(v) })
+      exempted = slice!(data, EXEMPTED_TAGS)
+
+      ResponseStruct.create(data.keys.map(&:to_sym) + exempted.keys.map(&:to_sym),
+                            data.values.map { |v| to_struct(v) } + exempted.values)
     end
 
     private
 
     attr_reader :required_attrs
 
+    def slice!(data, keys)
```

Comment:
could we make it clearer by changing `keys` -> `to_exclude` or something similar?

Reply:
How about `keys_to_select`?

Reply:
👍